src/decoder/stream.rs

```diff
@@ -99,7 +98,7 @@ pub enum DecodingError {
     /// number of calls
     ///
     /// If you're an application you might want to signal that a bug report is appreciated.
-    Parameter(Cow<'static, str>),
+    Parameter(ParameterError),
```

Comment:
Would it be better to call this DecodingError::InvalidParameter?

Reply:
I was trying to imitate the naming in `image`, mainly. I'm also not convinced the `invalid` qualifier adds anything substantial, it's an error so of course something wasn't right. The [API guidelines](https://rust-lang.github.io/api-guidelines/naming.html#names-use-a-consistent-word-order-c-word-order) recommend verb-object-error as a word order but there is no verb to apply here.

Reply:
You're probably right. Part of my confusion initially was from the variants of ParameterError: "Wrong data size" or "end of image has been reached" sound more like file corruption than something caused by passing a bad parameter.

Reply:
I can improve those of course :+1: